raiden/transfer/secret_registry.py

```diff
@@ -14,16 +20,28 @@ def events_for_onchain_secretreveal(
     secret: Secret,
     expiration: BlockExpiration,
     block_hash: BlockHash,
+    target_state: TargetTransferState = None,
+    pseudo_random_generator: random.Random = None
 ) -> List[Event]:
-    events: List[Event] = list()
-
     if not isinstance(secret, T_Secret):
         raise ValueError("secret must be a Secret instance")
 
     if get_status(channel_state) in CHANNEL_STATES_PRIOR_TO_CLOSED + (CHANNEL_STATE_CLOSED,):
-        reveal_event = ContractSendSecretReveal(
-            expiration=expiration, secret=secret, triggered_by_block_hash=block_hash
-        )
-        events.append(reveal_event)
+        if not channel_state.is_light_channel:
+            return [
+                ContractSendSecretReveal(
+                    expiration=expiration,
+                    secret=secret,
+                    triggered_by_block_hash=block_hash,
+                )
+            ]
+        if target_state is not None:
+            return [
+                ContractSendSecretRevealLight(
+                    message_id=message_identifier_from_prng(pseudo_random_generator),
```

Comment:
Why is a message_id needed here? messages_id are for offchain comms right?

Reply:
This is just a db requirement, for me its ok to keep it like that, but the msg id on onchain operations isnt too relevant